listenbrainz/webserver/views/user.py

```diff
@@ -180,6 +181,31 @@ def artists(user_name):
     )
 
 
+@user_bp.route('/<user_name>/delete-listen', methods=['POST'])
+@login_required
+def delete_listen():
+    """ Delete a particular listen from the currently logged-in user's listen history.
+
+    This checks for the correct authorization token and deletes the listen.
+    """
+    if request.form.get('token') and (request.form.get('token') == current_user.auth_token):
+        try:
+            recording_msid = request.form.get('recording_msid')
+            if recording_msid is None:
+                log_raise_400("%s Recording MSID missing.")
+
+            if not is_valid_uuid(recording_msid):
+                log_raise_400("%s MSID format invalid." % recording_msid)
+
+            delete_listen(user_name=current_user.musicbrainz_id, recording_msid=recording_msid)
+        except Exception as e:
```

Comment:
I think we can do this better -- the exception being caught is very general and we're sending the details of DB exception to the user, which isn't very nice. We should just tell them sorry, we could not delete this please try again later. I personally think you should catch the DB exception in the timescale module and raise something more generic that you can catch here. Make sure to log the error though.
If we can't delete listen, that is our fault so it should be a 500 error, not a 400. (remember 4xx is their fault, 5xx is our fault)

Reply:
I raised `TimescaleListenStoreException` and when its caught then return InternalServerError. If any other exception then raise BadRequest. Is it fine now?

Reply:
500 for general exceptions
503 for all Timescale exceptions

Reply:
yep